src/clear.c

```diff
@@ -151,35 +151,23 @@ static int clear_data (struct GMTAPI_CTRL *API, char *planet) {
 }
 
 static int clear_sessions (struct GMTAPI_CTRL *API) {
-	unsigned int n_dirs, k;
-	char **dirlist = NULL, *here = NULL;
+	int error;
+	char del_cmd[PATH_MAX] = {""};
 	if (access (API->session_dir, F_OK)) {
 		GMT_Report (API, GMT_MSG_INFORMATION, "No directory named %s\n", API->session_dir);
 		return GMT_FILE_NOT_FOUND;
 	}
-	if ((here = getcwd (NULL, 0)) == NULL) {	/* Get the current directory */
-		GMT_Report (API, GMT_MSG_ERROR, "Cannot determine current directory!\n");
+#ifdef WIN32
```

Comment:
My only doubt is this bloody macro. Never clearly understood the difference for ``_WIN32`` but I think thing both MinGW and VS define it, whilst Cygwin has a ``WIN32``. For playing safe I would prefer it to be ``_WIN32``